src/test/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterServiceUnitTests.java

```diff
@@ -0,0 +1,330 @@
+    replaceFilter.setSlurActionType(SlurActionType.REPLACE);
+
+    when(slurFilterRepository.findAll()).thenReturn(List.of(replaceFilter));
+
+    String resultText = slurFilterService.censorText(null);
+
+    verify(slurFilterRepository, times(1)).findAll();
+    assertNull(resultText, "Null was not returned when passed as a parameter");
+  }
+
+  @Test
+  void givenEmptyText_whenCensorText_thenReturnEmptyText()
+      throws SlurFilterReportException, SlurFilterBlockedException {
+
+    SlurFilter replaceFilter = new SlurFilter();
+    replaceFilter.setSlurRegex("\\bword.+?\\b");
+    replaceFilter.setSlurActionType(SlurActionType.REPLACE);
+
+    when(slurFilterRepository.findAll()).thenReturn(List.of(replaceFilter));
+
+    String resultText = slurFilterService.censorText("");
+
+    verify(slurFilterRepository, times(1)).findAll();
+    assertEquals("", resultText, "Returned text was not empty");
+  }
+
+  @Test
+  void givenBlankText_whenCensorText_thenReturnBlankText()
+      throws SlurFilterReportException, SlurFilterBlockedException {
+
+    SlurFilter replaceFilter = new SlurFilter();
+    replaceFilter.setSlurRegex("\\bword.+?\\b");
+    replaceFilter.setSlurActionType(SlurActionType.REPLACE);
+
+    when(slurFilterRepository.findAll()).thenReturn(List.of(replaceFilter));
+
+    String resultText = slurFilterService.censorText("        ");
+
+    verify(slurFilterRepository, times(1)).findAll();
+    assertEquals("        ", resultText, "Returned text was not blank");
+  }
+
+  @Test
+  void givenNoFiltersForText_whenCensorText_thenReturnText()
+      throws SlurFilterReportException, SlurFilterBlockedException {
+
+    when(slurFilterRepository.findAll()).thenReturn(Collections.emptyList());
+
+    String resultText = slurFilterService.censorText("text");
+
+    verify(slurFilterRepository, times(1)).findAll();
+    assertEquals("text", resultText, "Returned text was not blank");
+  }
+
+  @Test
+  void givenTextNeedingCensoring_whenCensorText_thenCensoredText()
+      throws SlurFilterReportException, SlurFilterBlockedException {
+
+    SlurFilter replaceFilter = new SlurFilter();
+    replaceFilter.setSlurRegex("mean");
+    replaceFilter.setSlurActionType(SlurActionType.REPLACE);
+
+    SlurFilter anotherReplaceFilter = new SlurFilter();
+    anotherReplaceFilter.setSlurRegex("\\bword.+?\\b");
+    anotherReplaceFilter.setSlurActionType(SlurActionType.REPLACE);
+
+    when(slurFilterRepository.findAll()).thenReturn(List.of(replaceFilter, anotherReplaceFilter));
+    when(slurFilterRepository.findBySlurActionType(SlurActionType.REPLACE)).thenReturn(List.of(replaceFilter, anotherReplaceFilter));
+
+    String resultText = slurFilterService.censorText("Some mean words in here.");
+
+    verify(slurFilterRepository, times(1)).findAll();
+    verify(slurFilterRepository, times(1)).findBySlurActionType(SlurActionType.REPLACE);
+
+    assertEquals("Some **** *********** in here.", resultText, "Censored text did not match expected");
```

Comment:
I noticed some odd behavior here that I am unsure is intentional. When censoring text the word getting censored ends up being replaced with a number of '*' equal to the length of the slur regex. In this case the word being censored has a length of 5, but after censoring is replaced by 11 '*'.

Reply:
That's odd... I'll poke around and see what I can find later tonight.

Reply:
I was able to recreate this issue. I'll create an issue to have it resolved.

Reply:
See issue [#262](https://github.com/sublinks/sublinks-api/issues/262)